src/installer.ts

```diff
@@ -123,8 +127,70 @@ function normalizeVersion(version: string): string {
     //append minor and patch version if not available
     return version.concat('.0.0');
   } else if (versionPart[2] == null) {
+    // handle beta and rc: 1.10beta1 => 1.10.0-beta1, 1.10rc1 => 1.10.0-rc1
+    if (versionPart[1].includes('beta') || versionPart[1].includes('rc')) {
+      versionPart[1] = versionPart[1]
+        .replace('beta', '.0-beta')
+        .replace('rc', '.0-rc');
+      return versionPart.join('.');
+    }
+
     //append patch version if not available
     return version.concat('.0');
   }
   return version;
 }
+
+async function determineVersion(version: string): Promise<string> {
+  if (!version.endsWith('.x')) {
+    return version;
+  }
+
+  return await getLatestVersion(version);
+}
+
+async function getLatestVersion(version: string): Promise<string> {
+  // clean .x syntax: 1.10.x -> 1.10
+  const trimmedVersion = version.slice(0, version.length - 2);
+
+  const versions = await getPossibleVersions(trimmedVersion);
+
+  core.debug(`evaluating ${versions.length} versions`);
+
+  if (version.length === 0) {
+    core.debug('match not found');
+    return trimmedVersion;
+  }
+
+  core.debug(`matched: ${versions[0]}`);
+
+  return versions[0];
+}
+
+interface IGoRef {
+  ref: string;
+}
+
+async function getAvailableVersions(): Promise<string[]> {
+  let rest: restm.RestClient = new restm.RestClient('setup-go');
+  let tags: IGoRef[] =
+    (await rest.get<IGoRef[]>(
+      'https://api.github.com/repos/golang/go/git/refs/tags'
+    )).result || [];
+
+  return tags
+    .filter(tag => tag.ref.match(/go\d+\.[\w\.]+/g))
+    .map(tag => tag.ref.replace('refs/tags/go', ''));
+}
+
+async function getPossibleVersions(version: string): Promise<string[]> {
+  const versions = await getAvailableVersions();
+  const possibleVersions = versions.filter(v => v.startsWith(version));
+
+  const versionMap = new Map();
```

Comment:
Rather than creating a map here, can you just do something like `return possibleVersions.sort()` (an alphabetic sort which matches the semver versioning ordering). I think what you're doing here is basically just doing that.

Reply:
What I am doing here is sorting using semver, which different with regular sort function. 

`sort` returns:

```
[ '1.10',
  '1.10.1',
  '1.10.2',
  '1.10.3',
  '1.10.4',
  '1.10.5',
  '1.10.6',
  '1.10.7',
  '1.10.8',
  '1.10beta1',
  '1.10beta2',
  '1.10rc1',
  '1.10rc2' ]
```

While this returns:

```
[ '1.10.8',
  '1.10.7',
  '1.10.6',
  '1.10.5',
  '1.10.4',
  '1.10.3',
  '1.10.2',
  '1.10.1',
  '1.10',
  '1.10rc2',
  '1.10rc1',
  '1.10beta2',
  '1.10beta1' ]
```

Reply:
Oh good point, sorry my mistake